### ISSUE TITLE
Update Form Controls Label, Required, Error, and Aria-Describedby

### ISSUE THREAD

**Issue by pdavies88** [department-of-veterans-affairs/va.gov-team#41785]
## Description
The following components need to be examined to make sure they are following the same format and practices listed below, and if they are not they need to be adjusted to be in compliance with the rules listed.
`va-date`, `va-number-input`, `va-text-input`, `va-radio`,  `va-radio-options`, `va-checkbox-group`, `va-checkbox`, `va-select`, `va-textarea`

Best Practices:
* Destructure all values set via this ie: `const {value, label} = this;` between render and return
* Label and Required should be in the following format:
```
        {label && (
          <label htmlFor="ID ON INPUT OR SELECT ELEMENT">
            {label}
            {required && (
              <span class="required">(*{i18next.t('required')})</span>
            )}
          </label>
        )}
```
* Error should be in the following format:
```
        {error && <span id="error-message" role="alert">
          <span class="sr-only">{i18next.t('error')}</span> {error}
          </span>
        }
```
* Aria-describedby should be in the following format and should not have a custom prop unless required by another team: 
```
aria-describedby={error ? 'error-message' : null}
```

## Acceptance Criteria
- [ ] Check all components listed above and make sure all are following the same guidelines via the best practices outlined above.
- [ ] Make any adjustments needed within the existing components to follow the guidelines
- [ ] Check and update the testing suites for the components with any changes made
- [ ] Check and update the CSS on the components for any updates made ex: `.sr-only` and `.required`
